contractcourt/htlc_timeout_resolver.go

```diff
@@ -391,7 +391,14 @@ func (h *htlcTimeoutResolver) spendHtlcOutput() (*chainntnfs.SpendDetail, error)
 		return nil, err
 	}
 
-	return h.waitForConfirmedSpend(outpointToWatch, scriptToWatch)
+	// If there's no mempool configured, which is the case for SPV node
+	// such as neutrino, then we will watch for confirmed spend only.
+	if h.Mempool == nil {
+		return h.waitForConfirmedSpend(outpointToWatch, scriptToWatch)
```

Comment:
If there is a preimage spend in the mempool, it will cause the resolver to clean up and mark it as `ResolverOutcomeClaimed`. But if the timeout tx eventually confirms, the resolver will not be around to mark it as `ResolverOutcomeTimeout`. This might mess up somebody's accounting. I think it would be better if the resolver wasn't cleaned up until an on-chain resolution occurs. We should be fine with sending multiple resolution messages for a single `CircuitKey` even if the types (success vs fail) differ. This newer behavior could also use an itest. If it's not possible to make an itest that we can commit, a hacky itest would also work that tests the behavior of duplicate resolution messages sent to the switch.

Reply:
IIUC, then below, even once we see the transaction in the mempool, we'll wait until things are fully confirmed before we try to clean up state. See the `consumeSpendEvents` method below. After we see things in the mempool we resolve then send back, but wait to exit the loop until things are confirmed on chain. 

Or do you mean that the incoming link should n't also be resolved (on disk) until the outgoing one? In the scenario where both links need to go on chain.

Reply:
> And does it also mean we can rely on the bucket contractsBucketKey to tell if an htlc is settled or failed?

I don't think so since when a contract (which includes success/timeout claims) is resolved, it is deleted from `contractsBucketKey`.